File: backend/content-service/src/main/java/com/educp/content/entity/SubColumn.java
package com.educp.content.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.educp.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@TableName("sub_columns")
@EqualsAndHashCode(callSuper = true)
public class SubColumn extends BaseEntity {

    private Long columnId;

    private String name;

    private String code;

    private String categoryCode;

    @TableField("is_system")
    private Boolean isSystem;

    private Integer sortOrder;

    private Long createdBy;
}

File: frontend/admin/src/types/subColumn.ts
export interface SubColumn {
  id: number
  columnId: number
  name: string
  code: string
  categoryCode?: string
  isSystem: boolean
  sortOrder: number
  createdAt: string
  updatedAt: string
}

export interface CreateSubColumnRequest {
  name: string
  code: string
  categoryCode?: string
  sortOrder?: number
}

export interface UpdateSubColumnRequest {
  name?: string
  code?: string
  categoryCode?: string
  sortOrder?: number
}

export interface SortSubColumnRequest {
  items: Array<{ id: number; sortOrder: number }>
}
